src/test/java/ch/uzh/ifi/hase/soprafs22/service/UserServiceTest.java

```diff
@@ -165,9 +263,413 @@ void updateUser_Conflict(){ //Test what happens when user tries to change userna
         putUser.setUsername("newUsername");
         putUser.setGender(Gender.MALE);
 
-        Mockito.when(userRepository.findByUsername(Mockito.any())).thenReturn(testUser);
+        Mockito.when(userRepository.findById(1L)).thenReturn(testUser);
+        Mockito.when(userRepository.findByUsername(putUser.getUsername())).thenReturn(conflictUser);
+
+        ResponseStatusException e = assertThrows(ResponseStatusException.class, () -> userService.updateUser(putUser));
+        assertEquals(HttpStatus.CONFLICT, e.getStatus());
+    }
+
+    @Test
+    void getWhiteCards_success_nonEmpty(){
+      //User has a White Card
+      testUser.setUserWhiteCards(testWhiteCards);
+      long id = testUser.getId();
+
+      Mockito.when(userRepository.findById(id)).thenReturn(testUser);
+
+      assertEquals(testWhiteCards, userService.getWhiteCards(id));
+    }
+
+    @Test
+    void getWhiteCards_success_Empty(){
+        //User does not have a White Card
+        long id = testUser.getId();
+
+        Mockito.when(userRepository.findById(id)).thenReturn(testUser);
+
+        assertEquals(new ArrayList<>(), userService.getWhiteCards(id));
+    }
+
+    @Test
+    void addGame_success(){
+      long id = testUser.getId();
+
+      Mockito.when(userRepository.findById(id)).thenReturn(testUser);
+
+      //should add the testgame to active games of testUser
+      userService.addGame(id, testGame);
+
+      assertEquals(testGame, testUser.getActiveGame());
+    }
+
+    @Test
+    void userHasNoActiveGame_true(){
+        long id = testUser.getId();
+
+        Mockito.when(userRepository.findById(id)).thenReturn(testUser);
+
+        //By default user shouldn't have an active game so should be true as no game added here
+        assertEquals(true, userService.userHasNoActiveGame(id));
+    }
+
+    @Test
+    void userHasNoActiveGame_false(){
+        long id = testUser.getId();
+
+        Mockito.when(userRepository.findById(id)).thenReturn(testUser);
+        //Game added to user
+        userService.addGame(id, testGame);
+        assertEquals(false, userService.userHasNoActiveGame(id));
```

Comment:
better use assertFalse here

Reply:
@JoMuelle was there a specific reason why you used assertEquals instead of assertFalse?